.github/actions/common-pre/action.yml

```diff
@@ -2,16 +2,29 @@ name: 'Common Test Setup Actions'
 runs:
   using: 'composite'
   steps:
-    - name: free disk space
-      if: matrix.os == 'ubuntu-latest'
-      uses: jlumbroso/free-disk-space@main
+    - name: Maximize build space (Linux)
+      if: runner.os == 'Linux'
+      uses: AdityaGarg8/remove-unwanted-software@v5
       with:
-        tool-cache: false
+        remove-dotnet: 'true'
+        remove-android: 'true'
+        remove-codeql: 'true'
+        remove-docker-images: 'true'
+
+    - name: Maximise build space (Mac OS)
+      if: runner.os == 'macOS'
+      shell: sh
+      run: |
+        sudo rm -rf /usr/share/dotnet
+        sudo rm -rf /opt/ghc
+        sudo rm -rf "/usr/local/share/boost"
+        sudo rm -rf "$AGENT_TOOLSDIRECTORY"
 
     - name: go toolchain
       uses: actions/setup-go@v5
       with:
         go-version: '=1.20.0'
+        cache: 'false'
```

Comment:
Not a Go project, we don't want it trying to cache

Reply:
Also... maybe we can delete this now?

Reply:
I tried but we're still using Go in the service check. We'd need to replace that before this can be removed